src/lib.rs

```diff
@@ -727,3 +727,211 @@ impl<'name, 'bufs, 'control> fmt::Debug for MsgHdrMut<'name, 'bufs, 'control> {
         "MsgHdrMut".fmt(fmt)
     }
 }
+
+/// Configuration of a `sendmmsg(2)` system call.
+///
+/// This wraps `mmsghdr` on Unix. Also see [`MmsgHdrMut`] for the variant used
+/// by `recvmmsg(2)`.
+#[cfg(all(
+    unix,
+    not(any(
+        target_os = "redox",
+        target_os = "solaris",
+        target_os = "hurd",
+        target_os = "vita",
+        target_os = "illumos",
+        target_vendor = "apple"
+    ))
+))]
+pub struct MmsgHdr<'addr, 'bufs, 'control> {
```

Comment:
What about matching `mmsghdr` here? For example 

```rust
pub struct MmsgHdr<'addr, 'bufs, 'control> {
    msg: MsgHdr,
    len: libc::c_uint,
}
```

This way the caller can decide if they want to use a `Vec`, a slice an array or something else.

Reply:
I see two options:

- go with your suggestion, and let callers provide their own slices of MmsgHdr. This works for the "lower level" `Socket::sendmmsg` function, but not for `Socket::send_multiple_to` as it needs some memory layout adaptation
- const generics may be an option? though we may still want to defer to the caller for that